charger/phoenix-em-eth.go

```diff
@@ -31,49 +31,24 @@ func init() {
 	registry.Add("phoenix-em-eth", NewPhoenixEMEthFromConfig)
 }
 
-//go:generate go run ../cmd/tools/decorate.go -f decoratePhoenixEMEth -b *PhoenixEMEth -r api.Charger -t "api.Meter,CurrentPower,func() (float64, error)" -t "api.MeterEnergy,TotalEnergy,func() (float64, error)" -t "api.PhaseCurrents,Currents,func() (float64, float64, float64, error)"
+//go:generate go run ../cmd/tools/decorate.go -f decoratePhoenixEMEth -b *PhoenixEMEth -r api.Charger -t "api.Meter,CurrentPower,func() (float64, error)" -t "api.MeterEnergy,TotalEnergy,func() (float64, error)" -t "api.PhaseCurrents,Currents,func() (float64, float64, float64, error)" -t "api.PhaseVoltages,Voltages,func() (float64, float64, float64, error)"
 
 // NewPhoenixEMEthFromConfig creates a Phoenix charger from generic config
 func NewPhoenixEMEthFromConfig(other map[string]interface{}) (api.Charger, error) {
-	cc := struct {
-		modbus.TcpSettings `mapstructure:",squash"`
-		Meter              struct {
-			Power, Energy, Currents bool
-		}
-	}{
-		TcpSettings: modbus.TcpSettings{
-			URI: "192.168.0.8:502", // default
-			ID:  180,               // default
-		},
+	cc := modbus.TcpSettings{
+		ID: 180,
 	}
 
 	if err := util.DecodeOther(other, &cc); err != nil {
 		return nil, err
 	}
 
-	wb, err := NewPhoenixEMEth(cc.URI, cc.ID)
```

Comment:
Könnten die Decorators in dieser Funktion bleiben? Dann liesse sich der "nackte" Controller auch ohne instanziieren.

Reply:
Ja, da stimmt ohnehin noch irgendwas mit der Dekoration nicht.
Muss ich mir nochmal näher ansehen.

Reply:
Das ist sonst auch nirgendwo so?
Welchem Zweck würde das dienen?

Reply:
> Das ist sonst auch nirgendwo so?

ja

> Dann liesse sich der "nackte" Controller auch ohne instanziieren.

Gibts einen Grund, das zu ändern?

Reply:
Angleichung an die anderen Implementierungen

Reply:
Bitte lass es so- auch andere Implementierungen nutzen den Weg wo es möglich ist.

Reply:
Ich verstehe noch nicht wie das hier genau funktionieren kann, da die Autodetection eine funktionierende Modbus-Verbindung zum Charger braucht. Aber wahrscheinlich stehe ich nur auf dem Schlauch und verstehe nur nicht was du mir sagen willst.
Hilfe! :) Lass uns am besten mal sprechen. Vielleicht geht mir dann ein Licht auf.

Reply:
Schau mal- ist einfach umgedreht.